[AVFoundation] remove shouldAddContentKeyRecipients code-path

jyavenard opened the pull request:
#### edd0064c6e44b70fc9c73d7f5c82fda039f0b24a
<pre>
[AVFoundation] remove shouldAddContentKeyRecipients code-path
<a href="https://bugs.webkit.org/show_bug.cgi?id=300270">https://bugs.webkit.org/show_bug.cgi?id=300270</a>
<a href="https://rdar.apple.com/162074955">rdar://162074955</a>

Reviewed by Jer Noble.

supportsAttachContentKey has been the default code-path since enabled in 272036@main on
iOS family and 282990@main all the other platforms.
The code was only partially removed in 299390@main. Remove the rest of it.
CDMSessionMediaSourceAVFObjC::add/removeParser were also unused. We removed them.

We can also remove the SPIs addStreamDataParser/removeStreamDataParser

No change in observable behaviour. Covered by existing tests.
* Source/WebCore/platform/graphics/avfoundation/AudioVideoRendererAVFObjC.mm:
(WebCore::AudioVideoRendererAVFObjC::addAudioRenderer):
(WebCore::AudioVideoRendererAVFObjC::destroyAudioRenderer):
(WebCore::AudioVideoRendererAVFObjC::ensureLayer):
(WebCore::AudioVideoRendererAVFObjC::destroyLayer):
(WebCore::AudioVideoRendererAVFObjC::setCDMInstance):
(WebCore::supportsAttachContentKey): Deleted.
(WebCore::shouldAddContentKeyRecipients): Deleted.
* Source/WebCore/platform/graphics/avfoundation/objc/CDMSessionAVContentKeySession.h:
(WebCore::CDMSessionAVContentKeySession::hasContentKeySession const):
* Source/WebCore/platform/graphics/avfoundation/objc/CDMSessionAVContentKeySession.mm:
(WebCore::CDMSessionAVContentKeySession::~CDMSessionAVContentKeySession):
(WebCore::CDMSessionAVContentKeySession::addParser): Deleted.
(WebCore::CDMSessionAVContentKeySession::removeParser): Deleted.
* Source/WebCore/platform/graphics/avfoundation/objc/CDMSessionMediaSourceAVFObjC.h:
* Source/WebCore/platform/graphics/avfoundation/objc/SourceBufferPrivateAVFObjC.mm:
(WebCore::SourceBufferPrivateAVFObjC::setCDMSession):
(WebCore::SourceBufferPrivateAVFObjC::attachContentKeyToSampleIfNeeded):
(WebCore::shouldAddContentKeyRecipients): Deleted.

Canonical link: <a href="https://commits.webkit.org/301167@main">https://commits.webkit.org/301167@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/d373299a4c01ca0905f2dd77aebcc14a084e60d9

| Misc | iOS, visionOS, tvOS & watchOS  | macOS  | Linux |  Windows | Apple Internal |
| ----- | ---------------------- | ------- |  ----- |  --------- | ------ |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/38/builds/124999 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/131/builds/44668 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/138/builds/35405 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/5/builds/131848 "Built successfully") | [✅ 🛠 win](https://ews-build.webkit.org/#/builders/59/builds/76973 "Built successfully") | ⏳ 🛠 ios-apple 
| [✅ 🧪 bindings](https://ews-build.webkit.org/#/builders/9/builds/126876 "Passed tests") | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/130/builds/45360 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/123/builds/53234 "Built successfully") | [✅ 🧪 wpe-wk2](https://ews-build.webkit.org/#/builders/34/builds/95144 "Passed tests") | [  ~~🧪 win-tests~~](https://ews-build.webkit.org/#/builders/60/builds/63193 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [✅ 🛠 mac-apple](https://ews-bridge.webkit.apple.com/builds/sw/T-276/92699e7b-e2c4-41dd-8b79-0fc7c00d5b52/d33d60d3-1fd9-4c0e-a49d-d4e76d76aa4e) 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/11/builds/127953 "Passed tests") | [✅ 🧪 ios-wk2](https://ews-build.webkit.org/#/builders/132/builds/36200 "Passed tests") | [✅ 🧪 api-mac](https://ews-build.webkit.org/#/builders/18/builds/111790 "Passed tests") | [✅ 🧪 api-wpe](https://ews-build.webkit.org/#/builders/41/builds/75687 "Passed tests") | | [✅ 🛠 vision-apple](https://ews-bridge.webkit.apple.com/builds/sw/T-276/92699e7b-e2c4-41dd-8b79-0fc7c00d5b52/a865e24f-f432-42c9-952b-6d103e0de6d0) 
| | [✅ 🧪 ios-wk2-wpt](https://ews-build.webkit.org/#/builders/133/builds/35124 "Passed tests") | [✅ 🧪 mac-wk1](https://ews-build.webkit.org/#/builders/135/builds/29943 "Passed tests") | [✅ 🛠 wpe-cairo](https://ews-build.webkit.org/#/builders/65/builds/75330 "Built successfully") | | 
| | [✅ 🧪 api-ios](https://ews-build.webkit.org/#/builders/13/builds/105963 "Passed tests") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/136/builds/30173 "Passed tests") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/2/builds/134524 "Built successfully") | | 
| | [✅ 🛠 vision](https://ews-build.webkit.org/#/builders/128/builds/51821 "Built successfully") | [✅ 🧪 mac-AS-debug-wk2](https://ews-build.webkit.org/#/builders/122/builds/39620 "Passed tests") | [✅ 🧪 gtk-wk2](https://ews-build.webkit.org/#/builders/1/builds/103616 "Passed tests") | | 
| | [✅ 🛠 vision-sim](https://ews-build.webkit.org/#/builders/121/builds/52246 "Built successfully") | [✅ 🧪 mac-wk2-stress](https://ews-build.webkit.org/#/builders/8/builds/108002 "Passed tests") | [✅ 🧪 api-gtk](https://ews-build.webkit.org/#/builders/21/builds/103390 "Passed tests") | | 
| [✅ 🛠 🧪 merge](https://ews-build.webkit.org/#/builders/19/builds/26352 "Built successfully and passed tests") | [✅ 🧪 vision-wk2](https://ews-build.webkit.org/#/builders/126/builds/48730 "Passed tests") | [✅ 🧪 mac-intel-wk2](https://ews-build.webkit.org/#/builders/137/builds/27020 "Passed tests") | [✅ 🛠 playstation](https://ews-build.webkit.org/#/builders/134/builds/48858 "Built successfully") | | 
| | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/127/builds/51706 "Built successfully") | [✅ 🛠 mac-safer-cpp](https://ews-build.webkit.org/#/builders/120/builds/57500 "Built successfully") | | | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/125/builds/51080 "Built successfully") | | | | 
| | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/129/builds/54437 "Built successfully") | | | | 
| | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/124/builds/52771 "Built successfully") | | | | 
<!--EWS-Status-Bubble-End-->

webkit-early-warning-system commented:
Starting EWS tests for https://github.com/WebKit/WebKit/commit/63f7bd474a5c35bd6bd6d91261e4098ab574b587. Live statuses available at the PR page, https://github.com/WebKit/WebKit/pull/51921

webkit-early-warning-system commented:
Starting EWS tests for https://github.com/WebKit/WebKit/commit/ee765164f07bbcc77b5af899381c1176703c2c25. Live statuses available at the PR page, https://github.com/WebKit/WebKit/pull/51921

webkit-early-warning-system commented:
Starting EWS tests for https://github.com/WebKit/WebKit/commit/d373299a4c01ca0905f2dd77aebcc14a084e60d9. Live statuses available at the PR page, https://github.com/WebKit/WebKit/pull/51921

jyavenard committed:
[AVFoundation] remove shouldAddContentKeyRecipients code-path
https://bugs.webkit.org/show_bug.cgi?id=300270
rdar://162074955

Reviewed by Jer Noble.

supportsAttachContentKey has been the default code-path since enabled in 272036@main on
iOS family and 282990@main all the other platforms.
The code was only partially removed in 299390@main. Remove the rest of it.
CDMSessionMediaSourceAVFObjC::add/removeParser were also unused. We removed them.

We can also remove the SPIs addStreamDataParser/removeStreamDataParser

No change in observable behaviour. Covered by existing tests.
* Source/WebCore/platform/graphics/avfoundation/AudioVideoRendererAVFObjC.mm:
(WebCore::AudioVideoRendererAVFObjC::addAudioRenderer):
(WebCore::AudioVideoRendererAVFObjC::destroyAudioRenderer):
(WebCore::AudioVideoRendererAVFObjC::ensureLayer):
(WebCore::AudioVideoRendererAVFObjC::destroyLayer):
(WebCore::AudioVideoRendererAVFObjC::setCDMInstance):
(WebCore::supportsAttachContentKey): Deleted.
(WebCore::shouldAddContentKeyRecipients): Deleted.
* Source/WebCore/platform/graphics/avfoundation/objc/CDMSessionAVContentKeySession.h:
(WebCore::CDMSessionAVContentKeySession::hasContentKeySession const):
* Source/WebCore/platform/graphics/avfoundation/objc/CDMSessionAVContentKeySession.mm:
(WebCore::CDMSessionAVContentKeySession::~CDMSessionAVContentKeySession):
(WebCore::CDMSessionAVContentKeySession::addParser): Deleted.
(WebCore::CDMSessionAVContentKeySession::removeParser): Deleted.
* Source/WebCore/platform/graphics/avfoundation/objc/CDMSessionMediaSourceAVFObjC.h:
* Source/WebCore/platform/graphics/avfoundation/objc/SourceBufferPrivateAVFObjC.mm:
(WebCore::SourceBufferPrivateAVFObjC::setCDMSession):
(WebCore::SourceBufferPrivateAVFObjC::attachContentKeyToSampleIfNeeded):
(WebCore::shouldAddContentKeyRecipients): Deleted.

Canonical link: https://commits.webkit.org/301167@main
